src/components/tree-item/tree-item.tsx

```diff
@@ -361,7 +361,8 @@ export class TreeItem
 
   @Listen("click")
   onClick(event: Event): void {
-    if (this.disabled) {
+    const isActionClick = (event.target as HTMLCalciteActionElement).slot === "actions-end";
```

Comment:
I think this is a fine solution but it would probably be better if the `onClick` was just removed from the `host` and set only onto the internal nodes that should allow the `onClick`.

Otherwise, as we add new slots, we would have to start filtering them out here. 

What do you think?

Reply:
If we do go this route, it might be better to use composedPath and check if the slot is in it.

Reply:
Yeah I think `@Listen("click"` is weird and kind of an anti-pattern. I can try to move this to be more targeted.

Reply:
I just looked at this a bit and it would be pretty substantial to rework this. It may also have accessibility impact. Let me try to use composedPath though, that way this will work in more cases.

Reply:
Updated. This will catch all clicks anywhere in the slotted dom and prevent selections.